### DISCUSSION TITLE
Long Print - Time Remaining is Very Wrong on Display

### DISCUSSION THREAD

**Discussion by clintiepoo** [Jyers/Marlin#1245] [Q&A] [answered]
I'm running a long print - it has about 31 hours remaining but the newest firmware says there is 11 hours remaining. I think this is a bug, maybe.

I'm less concerned about the percentage, but it would be nice if those also matched up...

Octoprint: 
![649387826 320981public](https://user-images.githubusercontent.com/33547246/127724657-293ae463-f98e-43e0-8f17-b8587227d521.JPEG)

Display: 
![649387828 977537public](https://user-images.githubusercontent.com/33547246/127724659-36e86fde-0eda-43ac-a47f-4753b875be42.jpg)

**Comment by clintiepoo** [comment]
Just following up again, here's another view of the remaining time being wrong.

Octoprint:
![20210731_123720632_iOS](https://user-images.githubusercontent.com/33547246/127740120-dbb1b2d7-40cd-40f6-80d1-576fd493a834.png)

Display: 
![649427866 025944public](https://user-images.githubusercontent.com/33547246/127740124-cba5ef58-d4cc-4998-b3aa-3e345e14904b.jpg)

**Reply by clintiepoo** [reply]
@Jyers have you seen this time remaining issue?

**Comment by Cal-LF** [comment]
This is not an issue. The time remaining is calculated by layer X time. It gets closer to reality as the print progresses.
Ref #1146

**Reply by clintiepoo** [reply]
I guess it seems like 19 hours is pretty different than 20 minutes. I can understand the linear approximation or whatever, but the values are so far off it seems like a bug. 

On the second picture, I'm at Z=56. That's layer 280. There are 929 layers in this print. 

Simple linear approximation would be: (17.78 hr / 280 layers) * 929 layers = 59 hours total. 59 - 17.78 = 41 hours remaining. This is wrong because the print is narrower on the top. But, I don't understand how the firmware can calculate 20 minutes when reality will be like 17 hours.

Interestingly enough, right now Octoprint says 17:50 remaining and the display says 17:06. It is getting closer.

**Comment by Jyers** [comment, marked as answer]
If you're running octoprint, the printer isn't calculating anything... Any progress/time being displayed by the printer is being sent to it by octoprint. Figure out whatever plugin is sending this data and then check to see why it's sending incorrect numbers. Personally, I use "M73 progress"
